src/plugins/discover/public/application/main/components/sidebar/discover_field_search.scss

```
@@ -2,6 +2,10 @@
   padding: $euiSizeM;
 }
 
-.dscFieldSearch__filterWrapper {
-  width: 100%;
+.dscFieldTypesHelp__popover {
+  flex-grow: 0;
 }
+
+.dscFieldTypesHelp__button {
+  @include euiFocusBackground($euiColorDarkShade);
```

Can you further explain what you mean by "I don't love how that looks" when using the gray?

If you're trying to match the append background color, it's not using the focus color but:
```suggestion
  background-color: $euiFormInputGroupLabelBackground;
  min-width: 0 !important; // Reduce width of icon-only button
```

I'd also recommend removing the min-width on the button so that it's only as wide as it needs to be (hence that second line ^^)